src/sql/src/parsers/alter_parser.rs

```diff
@@ -248,7 +233,40 @@ impl ParserContext<'_> {
                 } else if w.keyword == Keyword::SET {
                     // consume the current token.
                     self.parser.next_token();
-                    self.parse_alter_column_set_index(column_name)
+                    let ts = self.parser.peek_token();
+                    match ts.token {
+                        Token::Word(w) if w.keyword == Keyword::DEFAULT => {
+                            self.parser.prev_token();
+                            self.parser.prev_token();
+                            self.parser.prev_token();
+                            self.parser.prev_token();
+                            // Parse `SET DEFAULT`: ALTER TABLE `table_name` MODIFY COLUMN `a` SET DEFAULT 10, MODIFY COLUMN `b` SET DEFAULT "b", ...
+                            self.parse_alter_table_set_default()
+                        }
+                        Token::Word(w)
+                            if w.keyword == Keyword::FULLTEXT
+                                || w.value.eq_ignore_ascii_case(INVERTED)
+                                || w.value.eq_ignore_ascii_case("SKIPPING") =>
+                        {
+                            // Parse `DROP DEFAULT`: ALTER TABLE `table_name` MODIFY COLUMN `a` DROP DEFAULT, MODIFY COLUMN `b` DROP DEFAULT, ...
+                            self.parse_alter_column_set_index(column_name)
+                        }
+                        _ => self.expected("DEFAULT is expecting after SET", ts),
+                    }
+                } else if w.keyword == Keyword::DROP {
+                    // consume the current token.
+                    self.parser.next_token();
+                    let ts = self.parser.peek_token();
+                    match ts.token {
+                        Token::Word(w) if w.keyword == Keyword::DEFAULT => {
+                            self.parser.prev_token();
+                            self.parser.prev_token();
+                            self.parser.prev_token();
+                            self.parser.prev_token();
+                            self.parse_alter_table_drop_default()
+                        }
+                        _ => self.expected("DEFAULT is expecting after DROP", ts),
+                    }
```

Comment:
We can simplify the parser by:
```suggestion
                    self.parser.expect_keyword(Keyword::DEFAULT).context(error::SyntaxSnafu)?;
                    Ok(DropDefaultsOperation(column_name))
```